Tasks/Common/nuget-task-common/Utility.ts

```diff
@@ -82,7 +82,11 @@ export function resolveWildcardPath(pattern: string, allowEmptyWildcardMatch?: b
         // First find the most complete path without any matching patterns
         let idx = firstWildcardIndex(pattern);
         tl.debug("Index of first wildcard: " + idx);
-        let findPathRoot = path.dirname(pattern.slice(0, idx));
+
+        // include the wildcard character because:
+        //  dirname(c:\foo\bar\) => c:\foo (which will make find() return a bunch of stuff we know we'll discard)
+        //  dirname(c:\foo\bar\*) => c:\foo\bar
+        let findPathRoot = path.dirname(pattern.slice(0, idx + 1));
```

Comment:
If you want consistency, your team should be opinionated and remove almost all of the options from that interface. Consistent UI will just be even more confusing if each task selects different options underneath.

There should also be structures in the task definition to make writing the UX consistently be easier than not doing so.